app/src/main/java/org/wikipedia/dataclient/wikidata/Claims.kt

```diff
@@ -0,0 +1,101 @@
+package org.wikipedia.dataclient.wikidata
+
+import android.location.Location
+import com.google.gson.JsonElement
+import com.google.gson.annotations.SerializedName
+import org.wikipedia.dataclient.mwapi.MwResponse
+import org.wikipedia.json.GsonUtil
+
+class Claims : MwResponse() {
+
+    private val claims: Map<String, List<Claim>>? = null
+
+    fun claims(): Map<String, List<Claim>> {
+        return claims ?: emptyMap()
+    }
+
+    class Claim {
+
+        private val type: String? = null
+        private val id: String? = null
+        private val rank: String? = null
+        @SerializedName("mainsnak")
+        val mainSnak: MainSnak? = null
+    }
+
+    class MainSnak {
+
+        @SerializedName("snaktype")
+        private val snakType: String? = null
+        @SerializedName("datatype")
+        private val dataType: String? = null
+        private val property: String? = null
+        private val hash: String? = null
+        @SerializedName("datavalue")
+        val dataValue: DataValue? = null
+    }
+
+    class DataValue {
+
+        private val value: JsonElement? = null
+        private val type: String? = null
+
+        fun value(): String {
+            if (value != null) {
```

Comment:
This may be always `null` since you are using the `value` when the `DataValue` object is created. A proper way of doing this could be to put the logic to `get() { .... }`
```
val value: JsonElement? = null
     get() { .... }

```

Reply:
We are getting the string value from JSON @cooltey . Not sure how to simplify.

Reply:
Ah! Yes you are correct. My bad.